src/main/java/com/datadoghq/sketch/ddsketch/mapping/LogLikeIndexMapping.java

```diff
@@ -0,0 +1,116 @@
+package com.datadoghq.sketch.ddsketch.mapping;
+
+import java.util.Objects;
+
+/**
+ * A base class for mappings that are derived from a function that approximates the logarithm, namely {@link #log}.
+ * <p>
+ * That function is scaled depending on the targeted relative accuracy, the base of the logarithm that {@link #log}
+ * approximates and how well it geometrically pulls apart values from one another, that is to say, the infimum of
+ * |l(x)-l(y)|/|x-y| where x ≠ y and l = log∘exp (log being {@link #log}).
+ */
+abstract class LogLikeIndexMapping implements IndexMapping {
+
+    private final double relativeAccuracy;
+    private final double multiplier;
+    private final double normalizedIndexOffset;
+
+    LogLikeIndexMapping(double relativeAccuracy) {
+        if (relativeAccuracy <= 0 || relativeAccuracy >= 1) {
+            throw new IllegalArgumentException("The relative accuracy must be between 0 and 1.");
+        }
+        this.relativeAccuracy = relativeAccuracy;
+        this.multiplier = correctingFactor() * Math.log(base())
+            / (Math.log((1 + relativeAccuracy) / (1 - relativeAccuracy)));
+        this.normalizedIndexOffset = 0;
+    }
+
+    /**
+     * Constructs a mapping that approximates x -> log(x) + indexOffset, where log is to the base gamma.
+     *
+     * @param gamma       the base of the logarithm that the constructed mapping approaches
+     * @param indexOffset the value such that {@code logInverse(indexOffset / multiplier)} is the left bound of the
+     *                    bucket of index 0
+     */
+    LogLikeIndexMapping(double gamma, double indexOffset) {
+        if (gamma <= 1) {
+            throw new IllegalArgumentException("gamma must be greater than 1.");
+        }
+        this.relativeAccuracy = 1 - 2 / (1 + Math.exp(correctingFactor() * Math.log(gamma)));
+        this.multiplier = Math.log(base()) / Math.log(gamma);
+        this.normalizedIndexOffset = indexOffset - log(1) * this.multiplier;
+    }
+
+    /**
+     * @return an approximation of {@code log(1) + Math.log(value) / Math.log(base())}
+     */
+    abstract double log(double value);
+
+    /**
+     * The exact inverse of {@link #log}.
+     *
+     * @return the {@code value} such that {@code log(value) == index}
+     */
+    abstract double logInverse(double index);
+
+    /**
+     * @return the base of the logarithm that {@link #log} approaches
+     */
+    abstract double base();
+
+    /**
+     * @return a factor that corrects the fact that {@code log} may not geometrically pull apart values from one another
+     * as well as the logarithm
```

Comment:
can you add a couple words on how this is calculated? (the numbers look super-arbitrary in the classes :) )